File: 1894-find-the-student-that-will-replace-the-chalk/1894-find-the-student-that-will-replace-the-chalk.py
class Solution:
  def chalkReplacer(self, chalk: List[int], k: int) -> int:
    chalkSum = 0
    for val in chalk:
      chalkSum += val
    lastIterationChalkSum = k % chalkSum
    for i, val in enumerate(chalk):
      lastIterationChalkSum -= val
      if lastIterationChalkSum < 0:
        return i
    return 0
        

File: 1894-find-the-student-that-will-replace-the-chalk/1894-find-the-student-that-will-replace-the-chalk.go
func chalkReplacer(chalk []int, k int) int {
  chalkSum := 0;
  for _, val := range chalk {
    chalkSum += val;
  }
  lastIterationChalkSum := k % chalkSum;
  for i, val := range chalk {
    lastIterationChalkSum -= val;
    if lastIterationChalkSum < 0 {
      return i;
    }
  }
  return 0;
}